custom/enikshay/tasks.py

```diff
@@ -0,0 +1,267 @@
+from celery.schedules import crontab
+from celery.utils.log import get_task_logger
+from django.conf import settings
+from django.utils.dateparse import parse_datetime
+
+from casexml.apps.case.mock import CaseBlock
+from corehq import toggles
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.apps.fixtures.models import FixtureDataItem
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.apps.reports.analytics.esaccessors import get_reverse_indexed_cases_es
+from corehq.util.soft_assert import soft_assert
+from dimagi.utils.decorators.memoized import memoized
+
+from .case_utils import CASE_TYPE_ADHERENCE, CASE_TYPE_EPISODE
+
+
+DOSE_TAKEN_INDICATORS = [
+    'directly_observed_dose',
+    'unobserved_dose',
+    'self_administered_dose',
+]
+DOSE_MISSED = 'missed_dose'
+DOSE_UNKNOWN = 'missing_data'
+DOSE_KNOWN_INDICATORS = DOSE_TAKEN_INDICATORS + [DOSE_MISSED]
+DAILY_SCHEDULE_FIXTURE_NAME = 'adherence_schedules'
+DAILY_SCHEDULE_ID = 'schedule_daily'
+SCHEDULE_ID_FIXTURE = 'id'
+logger = get_task_logger(__name__)
+
+
+@periodic_task(
+    run_every=crontab(day_of_week=[1], hour=0, minute=0),  # every Monday
+    queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery')
+)
+def enikshay_adherence_task():
+    # runs adherence calculations for all domains that have `toggles.UATBC_ADHERENCE_TASK` enabled
+    domains = toggles.UATBC_ADHERENCE_TASK.get_enabled_domains()
+    for domain in domains:
+        updater = EpisodeAdherenceUpdater(domain)
+        updater.run()
+
+
+class EpisodeAdherenceUpdater(object):
+    """This iterates over all open 'episode' cases and sets 'adherence' related properties
+    according to this spec https://docs.google.com/document/d/1FjSdLYOYUCRBuW3aSxvu3Z5kvcN6JKbpDFDToCgead8/edit
+
+    This is applicable to various enikshay domains. The domain can be specified in the initalization
+    """
+
+    def __init__(self, domain):
+        self.domain = domain
+        self.purge_date = pytz.UTC.localize(datetime.datetime.today() - datetime.timedelta(days=60))
+
+    def run(self):
+        # iterate over all open 'episode' cases and set 'adherence' properties
+        for episode in self._get_open_episode_cases():
+            update = EpisodeUpdate(episode, self)
+            try:
+                case_block = update.case_block()
+                if case_block:
+                    submit_case_blocks(
+                        [ElementTree.tostring(case_block.as_xml())],
+                        self.domain
+                    )
+            except Exception, e:
+                logger.error("Error calculating adherence values for episode case_id({}): {}".format(
+                    episode.case_id,
+                    e
+                ))
+
+    def _get_open_episode_cases(self):
+        # return all open 'episode' cases
+        case_accessor = CaseAccessors(self.domain)
+        case_ids = case_accessor.get_open_case_ids_in_domain_by_type(CASE_TYPE_EPISODE)
```

Comment:
Should I switch this also to ES? This is just one time call per task run. 

I wasn't able to find ES util for this, it's pretty easy to write, I will write one if it's needed